### PR TITLE
Publish latency and failure metrics for cluster state applier thread

### PR DESCRIPTION
**Pull request by amathur1893**
Fixes #, if available: [564](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/issues/564)

Description of changes: With this change we will start publishing metrics like Latency and failure for Cluster state applier thread.

1.Tested using Docker
   
Tmp file

```
^cluster_applier_service
{"current_time":1614584101385}
{"ClusterApplierService_Failure":2,"ClusterApplierService_Latency":0}$
```

Table created

```
sqlite> .tables
ClusterApplierService_Latency 
ClusterApplierService_Failure
```

Contents of the table

```
sqlite> select * from ClusterApplierService_Failure;
6.0|6.0|6.0|6.0
sqlite> select * from ClusterApplierService_Latency;
29.0|29.0|29.0|29.0
```

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java [hunk 1]
```diff
@@ -821,6 +822,33 @@ public String toString() {
     }
   }
 
+  public enum ClusterApplierServiceStatsValue implements MetricValue {
+    /**
+     * Sum of total pending tasks throttled by master node.
+     */
```
Comment (nitinverma2510) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java:828]: Is this a copy-paste miss? The comment seems to be unrelated.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/model/MetricsModel.java [hunk 2]
```diff
@@ -374,6 +374,11 @@
         new MetricAttributes(
             MetricUnits.COUNT.toString(), AllMetrics.ShardStateDimension.values()));
 
+    allMetricsInitializer.put(
+            AllMetrics.ClusterApplierServiceStatsValue.CLUSTER_APPLIER_SERVICE_LATENCY.toString(),
+            new MetricAttributes(
+                    MetricUnits.MILLISECOND.toString(), EmptyDimension.values()));
+
```
Comment (nitinverma2510) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/model/MetricsModel.java:381]: Don't we need to include failure metric as well in here?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java [hunk 3]
```diff
@@ -42,8 +42,9 @@
     SHARD_STATS,
     MASTER_PENDING,
     MOUNTED_PARTITION_METRICS,
+    CLUSTER_APPLIER_SERVICE,
     ADMISSION_CONTROL_METRICS,
-    SHARD_INDEXING_PRESSURE
+    SHARD_INDEXING_PRESSURE;
```
Comment (yu-sun-77) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java:47]: Can we replace ";" with "," to avoid future changes when adding new metric?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/ClusterApplierService_Latency.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (yu-sun-77) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/ClusterApplierService_Latency.java:2]: Can we change year to 2021 in the license header?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java (hunk 3 above)
Comment (amathur1893) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java:47]: done

**Review by yu-sun-77 — APPROVED** [review]

**Review by nitinverma2510 — APPROVED** [review]
